power_grid_model_c/power_grid_model/include/power_grid_model/common/exception.hpp

```diff
@@ -112,6 +112,11 @@ class IterationDiverge : public PowerGridError {
     }
 };
 
+class MaxIterationReached : public PowerGridError {
```

Comment:
shouldn't this derive from `IterationDiverge`?

```suggestion
class MaxIterationReached : public IterationDiverge {
```

Reply:
After discussion: original `IterationDiverge` is probably not terribly accurate, but for the sake of keeping minimum change to the user side, `MaxIteration` error should derive from it and only mention the maximum iteration reached situation. This has nothing to do with convergence per se.